### PR TITLE
Add strict null checks to @pixi/ticker

### PR DESCRIPTION
**Pull request by 8times12**
##### Description of change

Ref https://github.com/pixijs/pixijs/issues/8852

I have made the following adjustments to resolve the strict null check errors occurring under the packages/ticker directory.

[packages/ticker/src/TickerPlugin.ts](https://github.com/pixijs/pixijs/pull/10087/files#diff-2dd43e1902cf45a5c2f73ba17c20286886683f9906e240e9091cf47b37e6a743)

- Made `_ticker` and `ticker` nullable and added optional chaining when referenced.

[packages/ticker/src/TickerListener.ts](https://github.com/pixijs/pixijs/pull/10087/files#diff-e1e288f1edb157d67a51a66dde73bb2ee0420e9fb9ab355f8a4ec170fc29c515)

- Due to a circular reference issue, defined `TickerCallback` within this file.
- In the `emit` method, to enable type inference for the `fn` property, set the type of TickerCallback to a union type.
- To make the type definition more strict, changed the type argument `T` of the class `TickerListener` from `any` to `Record<string, any> | null`.
  ```diff
  - export class TickerListener<T = any>
  + TickerListenerContext = Record<string, any> | null
  + export class TickerListener<T extends TickerListenerContext = TickerListenerContext>
  ```
- Integrated the definitions of the properties `fn`, `context`, `priority`, and `once` with the constructor's parameter definitions. This prevents redundant type management between the properties and constructor parameters. Additionally, since `null` is set to `fn` and `context` within the destroy method, made them nullable.
  ```diff
  - public priority: number;
  - private fn: TickerCallback<T>;
  - private context: T;
  - private once: boolean;
  - constructor(fn: TickerCallback<T>, context: T = null, priority = 0, once = false)
  - {
  -     this.fn = fn;
  -     this.context = context;
  -     this.priority = priority;
  -     this.once = once;
  + constructor(
  +     private fn: TickerCallback<T> | null,
  +     private context: T | null = null,
  +     public priority = 0,
  +     private once = false)
  ```

[packages/ticker/src/Ticker.ts](https://github.com/pixijs/pixijs/pull/10087/files#diff-8370bd094408139eddb3b1053b8031f0fe189c99d7c710cd64665ecd92ef84cc)

- Since `null` is set to `_head` and `_requestId` within the destroy method, made them nullable.
- Added null check processing for the nullable `_head`.

##### Pre-Merge Checklist

- [x] Tests and/or benchmarks are included
- [x] Documentation is changed or added
- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

### CODE REVIEW

**Comment by 8times12** [comment]
The CI tests are failing in packages/app/test/Application.tests.ts, which is unrelated to the changes made this time. However, they pass when executed locally.

File: packages/ticker/src/Ticker.ts [hunk 1]
```diff
@@ -419,6 +439,11 @@ export class Ticker
             // during the emit, we can still check for head.next
             const head = this._head;
 
+            if (!head)
+            {
+                return;
+            }
```
Comment (crllrd) [inline packages/ticker/src/Ticker.ts:445]: Maybe it make sense to  check this.head, and only after create a variable ?

```suggestion
            if (!this._head)
            {
                return;
            }
            const head = this._head;
```

File: packages/ticker/test/TickerPlugin.tests.ts [hunk 2]
```diff
@@ -4,25 +4,25 @@ describe('TickerPlugin', () =>
...
     it('should not start application before calling start method if options.autoStart is false', (done) =>
     {
-        const appp = {} as App;
+        const appp: App = {};
 
         TickerPlugin.init.call(appp, { autoStart: false });
 
         expect(appp.ticker).toBeInstanceOf(Ticker);
-        expect(appp.ticker.started).toBe(false);
+        expect(appp.ticker?.started).toBe(false);
```
Comment (crllrd) [inline packages/ticker/test/TickerPlugin.tests.ts:21]: Sorry, maybe I don't get this part, but it feels this 2 lines have conflicting logic, no ?

**Comment by Zyie** [comment]
Hey,

Thank you for putting together this PR! Since we've moved to the next major version of the library, we're closing this to focus on the updated codebase. If this is still relevant, feel free to submit a new PR for the latest version.

**Pull request closed without merging**